optuna/pruners/_hyperband.py

```diff
@@ -152,6 +158,12 @@ def __init__(
                 "But max_resource = {}".format(self._max_resource)
             )
 
+        if self._bootstrap_count > 0 and self._max_resource == "auto":
+            raise ValueError(
+                "bootstrap_count > 0 and max_resource == 'auto' "
+                "are mutually incompatible, bootstrap_count is {}".format(self._bootstrap_count)
+            )
+
```

Comment:
I was thinking that, since `bootstrap_count` passed to `SuccessiveHalvingPruner` it should be checked there to keep the checks in one case (so that if, for some reason, this restriction is lifted the hyperband pruner is transparently using the new strategy), but I can also add a check here. I will also look into the test cases.

Reply:
I agree with you. Could you add test cases?